dagshub/data_engine/model/datasource.py

```diff
@@ -61,13 +61,20 @@ class DatapointMetadataUpdateEntry(json.JSONEncoder):
     allowMultiple: bool = False
 
 
+@dataclass
+class Field:
+    column: str
+    as_of_timestamp: Optional[float] = None
+    alias: Optional[str] = None
```

Comment:
This needs to handle both raw timestamps AND datetime objects

Right now it looks like the **only** way to have an as_of right now is to filter directly with the timestamp, which is certainly not user-friendly. We should be able to handle this:
```
as_of_date = datetime.now() - timedelta(days=1)
ds.select(Field("blabla", as_of=as_of_date))
```

How to implement:
Basically the `as_of` needs to be a `Optional[Union[float, datetime]]` and the `as_of_timestamp` will be a property that will have special sauce to turn the `datetime` into a float.